R/Lrnr_glm_semiparametric

```diff
@@ -0,0 +1,251 @@
+#'
+#' @docType class
+#'
+#' @importFrom R6 R6Class
+#'
+#' @export
+#'
+#' @keywords data
+#'
+#' @return A learner object inheriting from \code{\link{Lrnr_base}} with
+#'  methods for training and prediction. For a full list of learner
+#'  functionality, see the complete documentation of \code{\link{Lrnr_base}}.
+#'
+#' @format An \code{\link[R6]{R6Class}} object inheriting from
+#'  \code{\link{Lrnr_base}}.
+#'
+#' @family Learners
+#'
+#' @section Parameters:
+#' \describe{
+#'   \item{\code{formula_sp}}{ A \code{\link{formula}} object specifying the parametric component of the semiparametric model.
+#'   }
+#'   \item{\code{lrnr_baseline}}{A baseline learner for estimation of the nonparametric component. This can be pooled or unpooled by specifying \code{return_matrix_predictions}}
+#'   \item{\code{interaction_variable = "A"}}{A interaction variable name (that can be found in `training_task$data`) to multiply by the design matrix generated by \code{formula_sp}. If NULL then the interaction variable is treated identically `1`.
+#'   In many applications, this will be the name of a binary treatment variable (e.g. `A`).
+#'   }
+#'   \item{\code{family = NULL}}{A family object whose link function specifies the type of semiparametric model (e.g. partially-linear least-squares (\code{\link{gaussian}}), partially-linear logistic regression (\code{\link{binomial}}), partially-linear log-linear regression (\code{\link{poisson}})
+#'    }
+#'   \item{\code{append_interaction_matrix = TRUE}}{Whether \code{lrnr_baseline} should be fit on `cbind(task$X,A*V)` where `A` is the interaction variable and `V` is the design matrix obtained from \code{formula_sp}.
+#'   Note, if `append_interaction_matrix = TRUE`, the resulting estimator will be projected onto the semiparametric model using \code{glm.fit}.
+#'   If this is FALSE and \code{interaction_variable} is binary then the semiparametric model is learned by stratifying on \code{interaction_variable}.
+#'   Specifically, if FALSE, \code{lrnr_baseline} is used to estimate `E[Y|A=0,W]` by subsetting to only observations with `A = 0`.
+#'   In the binary case, setting `append_interaction_matrix = TRUE` allows one to pool the learning across treatment arms and allows additive models to perform well.
+#'   }
+#'   \item{\code{return_matrix_predictions = FALSE}}{Only used if \code{interaction_variable} is binary. Whether to return a matrix output with three columns being `E[Y|A=0,W], E[Y|A=1,W], E[Y|A,W]`.
+#'   }
+#'  \item{\code{...}}{Not used.
+#'  }
+#' }
+#'
+#' @examples
+#' library(glmnet)
+#' n <- 200
+#' W <- runif(n, -1, 1)
+#' A <- rbinom(n, 1, plogis(W))
+#' Y_continuous <- rnorm(n, mean = A + W, sd = 0.3)
+#' Y_binary <- rbinom(n, 1, plogis(A + W))
+#' Y_count <- rpois(n, exp(A + W))
+#' data <- data.table(W, A, Y_continuous, Y_binary, Y_count)
```

Comment:
It's great to see so many examples. Note, however, that they need to all run in under 5sec (per CRAN policy), so consider wrapping some of them in `\dontrun{...}` blocks if the timing is cumulatively beyond the 5sec limit.